### ISSUE TITLE
Dependency to "curl"

### ISSUE THREAD

**Issue by pkordylewski** [mayflower/puppet-php#76]
Hey,

i know this should be really unlikely but i have a machine without "curl" installed. So i got:

```
Error: Could not find command 'curl'
Error: /Stage[main]/Php::Repo::Debian/Exec[add_dotdeb_key]/returns: change from notrun to 0 failed: Could not find command 'curl'
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/File[source_php_wheezy-php56.list]: Dependency Exec[add_dotdeb_key] has failures: true
Warning: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/File[source_php_wheezy-php56.list]: Skipping because of failed dependencies
```

Cheers,
Patryk

**Comment by mmoll** [comment]
@fpletz @globin anything against using ```apt::key``` from ```puppetlabs-apt``` here?

**Comment by fpletz** [comment]
@mmoll Full ack, `apt::key` should be used here. I really wonder why we don't do this yet.

**Comment by mmoll** [comment]
@pkordylewski could you test this?
```diff
diff --git a/manifests/repo/debian.pp b/manifests/repo/debian.pp
index 58f1e5c..3288dba 100644
--- a/manifests/repo/debian.pp
+++ b/manifests/repo/debian.pp
@@ -14,6 +14,9 @@
 # [*include_src*]
 #   Add source source repository
 #
+# [*key*]
+#   Public key in apt::key format
+#
 # [*dotdeb*]
 #   Enable special dotdeb handling
 #
@@ -22,7 +25,8 @@ class php::repo::debian(
   $release      = 'wheezy-php55',
   $repos        = 'all',
   $include_src  = false,
-  $dotdeb       = true
+  $dotdeb       = true,
+  $key          = { 'id' => '7E3F070089DF5277', 'source' => 'http://www.dotdeb.org/dotdeb.gpg' }
 ) {
 
   if $caller_module_name != $module_name {
@@ -36,6 +40,7 @@ class php::repo::debian(
     release     => $release,
     repos       => $repos,
     include_src => $include_src,
+    key         => $key,
   }
 
   if ($dotdeb) {
@@ -49,13 +54,5 @@ class php::repo::debian(
         include_src => $include_src,
       }
     }
-
-    exec { 'add_dotdeb_key':
-      command => 'curl -L --silent "http://www.dotdeb.org/dotdeb.gpg" | apt-key add -',
-      unless  => 'apt-key list | grep -q dotdeb',
-      path    => [ '/bin/', '/sbin/' , '/usr/bin/', '/usr/sbin/' ];
-    }
-
-    Exec['add_dotdeb_key'] -> ::Apt::Source["source_php_${release}"]
   }
 }
```

**Comment by mmoll** [comment]
@pkordylewski ping?

**Comment by pkordylewski** [comment]
Sorry, i will try to test the fix next week!

**Comment by MostAwesomeDude** [comment]
Tested and doesn't work for me. However, I'm quite bad at Puppet and perhaps I've missed something.

    Error: validate_re(): {"id"=>"7E3F070089DF5277", "source"=>"http://www.dotdeb.org/dotdeb.gpg"} does not match ["\\A(0x)?[0-9a-fA-F]{8}\\Z", "\\A(0x)?[0-9a-fA-F]{16}\\Z", "\\A(0x)?[0-9a-fA-F]{40}\\Z"] at /etc/puppet/modules/apt/manifests/key.pp:62 on node example.com

**Comment by mmoll** [comment]
ah, this is dependent on the puppetlabs-apt version.
@fpletz @globin anything against a PR switching to puppetlabs-apt 2.0?

**Comment by globin** [comment]
I'd like to keep puppetlabs-apt 1.* for now since 2.1 enables a compatibility layer which only throws deprecation warnings until more people have switched to 2.*

**Comment by fpletz** [comment]
This was fixed with in 5c264fab54261ce0fcfdb7b436e871f36534d4ad & bfc48d633268d201081e3faf80e9159f9ee13423.
